docs/madr/decisions/049-policy-on-namespace.md

```diff
@@ -0,0 +1,681 @@
+                tags:
+                  version: v2
+---
+# zone-1, producer policy
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+  name: backend-route-timeout
+  namespace: backend-ns
+  labels:
+    kuma.io/zone: zone-1
+spec:
+  to:
+    - targetRef:
+        kind: MeshHTTPRoute # doesn't work like this today, we need https://github.com/kumahq/kuma/issues/10247
+        name: backend-route
+      default:
+        http:
+          requestTimeout: 5s
+```
+
+```yaml
+# zone-2, consumer policy
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+  name: frontend-to-backend
+  namespace: frontend-ns
+  labels:
+    kuma.io/zone: zone-2
+spec:
+  to:
+    - targetRef:
+        kind: MeshHTTPRoute # technically we can use 'name' and 'namespace' but we'd have to know hashed name of the route after syncing
+        labels:
+          kuma.io/display-name: backend-route
+          k8s.kuma.io/namespace: backend-ns
+          kuma.io/zone: zone-1
+      default:
+        http:
+          requestTimeout: 3s
+```
+
+### Referencing MeshGateway
+
+`MeshGateway` at this moment is not namespace-scoped. It can change in the future with
+the [MeshBuiltinGateway resource](https://github.com/kumahq/kuma/issues/10014).
+
+For policies in custom namespaces, we've decided that we would allow targeting `MeshGateway` in topLevel targetRef in 
+`MeshHTTPRoute` and `MeshTCPRoute`, and policies that configures only Envoy clusters (policies that allows configuring: `to[].targetRef.kind: MeshService` like `MeshCircuitBreaker`). 
+For other policies we would allow referencing Mesh*Routes that reference `MeshGateway` in topLevel targetRef.
+
+#### Examples
+
+As a **Cluster Operator** I want to deploy a single global `MeshGateway` to the cluster so that I’m able to manage a
+single deployment for the whole cluster without knowing specific details about apps and routes.
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshGateway
+mesh: default
+metadata:
+  name: edge-gateway
+  labels:
+    kuma.io/origin: zone
+spec:
+  conf:
+    listeners:
+      - port: 80
+        protocol: HTTP
+  selectors:
+    - match:
+        kuma.io/service: edge-gateway
+```
+
+##### Targeting MeshGateway in MeshHttpRoute
+
+As a **Frontend Service Owner** I want to configure routes on `MeshGateway` without the access to
+cluster-scoped CRDs so that I could manage my routes independently.
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshHTTPRoute
+metadata:
+  name: http-route-1
+  namespace: frontend-ns
+  labels:
+    kuma.io/origin: zone-1
+spec:
+  targetRef:
+    kind: MeshGateway
+    name: edge-gateway
+  to:
+    - targetRef:
+        kind: Mesh
+      rules:
+        - matches:
+            - path:
+                type: PathPrefix
+                value: /gui
+          default:
+            backendRefs:
+              - kind: MeshService
+                name: frontend
+                # namespace: frontend-ns - is added automatically
+```
```

Comment:
Isn't there an access issue here? Anyone, anywhere could break the gateway no?

Feels like we should not allow referencing MeshGateway from namespaced policies ftm.

Reply:
What do you mean by breaking the gateway? I see we don't validate `backendRefs` at this moment and potentially you can redirect traffic to any namespace in the cluster. Is it what you meant by the access issue?

Reply:
I'm not sure we should have namespaced HTTPRoutes attach to MeshGateway either. Why should frontend be able to decide whether some arbitrary traffic makes it to their service?

For outbound policies though I think there's no issue, since it's already been decided that the traffic is headed to their service.

Reply:
right, that makes sense, I've changed the MADR accordingly